src/main/scala/com.socrata.snapshotter/BlobManager.scala

```diff
@@ -0,0 +1,49 @@
+package com.socrata.snapshotter
+
+import java.io.InputStream
+
+import com.amazonaws.event.{ProgressListener, ProgressEvent}
+import com.amazonaws.services.s3.AmazonS3Client
+import com.amazonaws.services.s3.model.{ObjectMetadata, PutObjectRequest}
+import com.amazonaws.services.s3.transfer.TransferManager
+import com.amazonaws.services.s3.transfer.model.UploadResult
+
+import com.rojoma.json.v3.interpolation._
+import com.rojoma.json.v3.ast.JValue
+
+import org.slf4j.LoggerFactory
+
+object BlobManager {
+  private lazy val s3client = new AmazonS3Client()
+  private lazy val manager = new TransferManager(s3client)
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  def shutdownManager(): Unit = {
+    logger.info("Shutting down transfer manager.")
+    manager.shutdownNow()
+  }
+
+  def upload(inStream: InputStream, keyName: String): UploadResult = {
```

Comment:
Didn't realize you owned this code.

Merging the other `upload` method into this is fine, and I think would actually be less confusing.